doc/package-notes.xml

```diff
@@ -704,6 +704,42 @@ overrides = super: self: rec {
 }; }
 </programlisting>
   </para>
+  <para>
+    WeeChat allows to set defaults on startup using the <literal>--run-command</literal>.
+    The <literal>configure</literal> method can be used to pass commands to the program:
+<programlisting>weechat.override {
+  configure = { availablePlugins, ... }: {
+    plugins = builtins.attrValues availablePlugins;
+
+    init = ''
+      /set foo bar
+      /server add freenode chat.freenode.org
+      /even
+      /more
+      /commands
+    '';
+  };
+}</programlisting>
+    These values can be overridden manually when running
+    <literal>weechat --run-command "your-commands"</literal>.
+  </para>
+  <para>
+    The addition of the <literal>init</literal> parameter to the <literal>override</literal> API
+    of WeeChat allows <literal>Nix</literal> to generate packages that autoconfigure themselves.
+    Furthermore it's possible to declaratively specify scripts to be loaded on startup:
+<programlisting>weechat.override {
+  configure = { availablePlugins, ... }: {
+    plugins = builtins.attrValues availablePlugins;
+
+    scripts = [
+      "${pkgs.weechat-xmpp}/share/jabber.py"
```

Comment:
Yeah something like

```
scripts  = with pkgs.weechatScripts [
  xmpp
  matrix-protocol
]
```

Would be pretty neat. It would be possible to remain backwards compatibility, although I'm not sure if its worth it.

Reply:
I'm not sure, but I'd even say that this is out of scope. This PR already includes script/config support for the package *and* a module and I'd say that splitting single scripts into their own metapackage should be done in a second PR.

Regarding the change itself: I'm definetely in favor of it, users don't need to know about the structure inside a given store path, they simply need to know their packages, furthermore this is way more readable IMHO.

Regarding backwards compatibility: I've seen several people using these packages, so I think that just moving them is a bad idea. Let's place them into `pkgs/top-level/aliases.nix` and print a deprecation warning if this attribute path is used.

Reply:
Scripts outside of the nix store would be an impurity that would make the config less portable. I see how it could be useful however, so if you want to support that its fine by me.

Reply:
I guess you're right, packaging weechat scripts shouldn't be too hard.I'll see how much time it costs to implement this, then I may add this to this PR.